lib/AST/ParameterPack.cpp

```diff
@@ -193,6 +193,15 @@ TupleType *TupleType::flattenPackTypes() {
   if (!anyChanged)
     return this;
 
+  // If pack substitution yields a single-element tuple, the tuple
+  // structure is flattened to produce the element type.
+  if (elts.size() == 1) {
+    auto type = elts.front().getType();
+    if (!type->is<PackExpansionType>() && !type->is<TypeVariableType>()) {
```

Comment:
I think we should reorganize things so that AST can check type variable flags, then we can look at TVO_CanBindToPack here. There was another place I wanted that recently too

Reply:
Yeah, that was the only reason I needed to add a very similar bit of code to `simplifyTypeImpl`